engine.py

```diff
@@ -161,114 +160,72 @@ def log_error(self, msg):
         self._log.error(msg)
 
     ##########################################################################################
-    # pyside / qt
-    
-    def _define_qt_base(self):
-        """
-        check for pyside then pyqt
-        """
+    # qt
+
+    def _define_unavailable_base(self):
         # proxy class used when QT does not exist on the system.
         # this will raise an exception when any QT code tries to use it
-        class QTProxy(object):                        
+        class QTProxy(object):
             def __getattr__(self, name):
                 raise tank.TankError("Looks like you are trying to run an App that uses a QT "
                                      "based UI, however the Shell engine could not find a PyQt "
-                                     "or PySide installation in your python system path. We " 
+                                     "or PySide installation in your python system path. We "
                                      "recommend that you install PySide if you want to "
                                      "run UI applications from the Shell.")
+        return {
+            "qt_core": QTProxy(),
+            "qt_gui": QTProxy(),
+            "qt_widgets": QTProxy(),
+            "dialog_base": None
+        }
+
+    def _define_qt_base(self):
+        """
+        check for pyside then pyqt
+        """
+        base = Engine._define_qt_base(self)
+
+        # If QtCore hasn't been set, then nothing was.
+        if base["qt_core"] is None:
+            return self._define_unavailable_base()
+
+        try:
+            QtCore = base["qt_core"]
+            DialogBase = base["dialog_base"]
+            QtWrapper = base["wrapper"]
+
+            QtCore.QTextCodec.setCodecForCStrings(
+                QtCore.QTextCodec.codecForName("utf-8")
+            )
+
+            # a simple dialog proxy that pushes the window forward
+            class ProxyDialog(DialogBase):
+                def show(self):
+                    DialogBase.show(self)
+                    self.activateWindow()
+                    self.raise_()
+
+                def exec_(self):
+                    self.activateWindow()
+                    self.raise_()
+                    # the trick of activating + raising does not seem to be enough for
+                    # modal dialogs. So force put them on top as well.
+                    self.setWindowFlags(QtCore.Qt.WindowStaysOnTopHint | self.windowFlags())
+                    return DialogBase.exec_(self)
+
+            base["dialog_base"] = ProxyDialog
+
+            self._has_qt = True
+            # Strip the name since PySide has a \n at the end for some reason.
+            self.log_debug("Successfully initialized %s '%s' located in %s." %
+                           (QtWrapper.__name__.strip(), QtWrapper.__version__, QtWrapper.__file__))
```

Comment:
are these unified in the base class now? Seems to be quite a few variations of how the version is reported back in pyside/pyqt etc.

Reply:
I believe this is addressed by tk-core's `util.QtImporter`.